### PR TITLE
feat: Prepare community operators scripts to add new bundles format

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
This PR prepare the community operators to have new bundles format

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19244

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by flacatus** [comment]
We can merge after 7.35.2

**Review by mmorhun — APPROVED** [review]
LGTM

File: olm/prepare-community-operators-update.sh [hunk 1]
```diff
@@ -132,39 +134,32 @@ do
...
     -e "/^  version: ${lastPackagePreReleaseVersion}/i\ \ replaces: eclipse-che.v${lastPublishedPackageVersion}" \
     -e "s/${packageName}/eclipse-che/" \
     "${packageBaseFolderPath}/${lastPackagePreReleaseVersion}/che-operator.clusterserviceversion.yaml" \
-    > "${folderToUpdate}/${lastPackagePreReleaseVersion}/eclipse-che.v${lastPackagePreReleaseVersion}.clusterserviceversion.yaml"
+    > "${folderToUpdate}/${lastPackagePreReleaseVersion}/manifests/eclipse-che.v${lastPackagePreReleaseVersion}.clusterserviceversion.yaml"
 
     echo
+
     echo "   - Update the CRD files"
     cp "${packageBaseFolderPath}/${lastPackagePreReleaseVersion}/org_v1_che_crd.yaml" \
```
Comment (AndrienkoAleksandr) [inline olm/prepare-community-operators-update.sh:149]: Minor and up to you, but I think we can make this 'cp' command without new line delimiter `\`

**Comment by AndrienkoAleksandr** [comment]
Hello, @flacatus, @tolusha. I see in the community update script we are using migration from new bundle to old format bundle command: https://github.com/eclipse-che/che-operator/blob/86d748bee0de01756fd503bf418b677f4f6283be/olm/prepare-community-operators-update.sh#L68  Then we recopy files from old format to follow new bundle format file locations. Maybe in the frame of the issue: we should remove it https://github.com/eclipse/che/issues/19244 and copy bundle from https://github.com/eclipse-che/che-operator/tree/86d748bee0de01756fd503bf418b677f4f6283be/bundle folder ? Or Did I misunderstand something?

File: olm/prepare-community-operators-update.sh (hunk 1 above)
Comment (flacatus) [reply olm/prepare-community-operators-update.sh:149]: +1

**Comment by flacatus** [comment]
> https://github.com/eclipse-che/che-operator/tree/86d748bee0de01756fd503bf418b677f4f6283be/bundle

Make sense to copy from https://github.com/eclipse-che/che-operator/tree/86d748bee0de01756fd503bf418b677f4f6283be/bundle

File: olm/prepare-community-operators-update.sh [hunk 2]
```diff
@@ -132,38 +135,29 @@ do
     echo $platform
     echo "     => will create release '${lastPackagePreReleaseVersion}' in the following package folder :'${folderToUpdate}'"
 
-    mkdir -p "${folderToUpdate}/${lastPackagePreReleaseVersion}"
+    mkdir -p "${folderToUpdate}/${lastPackagePreReleaseVersion}/{manifests,metadata}"
+    mkdir -p "${folderToUpdate}/${lastPackagePreReleaseVersion}/metadata"
```
Comment (AndrienkoAleksandr) [inline olm/prepare-community-operators-update.sh:139]: I think we can remove this line, cause we've created metadata folder in the line upper.

**Comment by flacatus** [comment]
Maybe we can move the function:
```bash
source ${OPERATOR_REPO}/.github/bin/common.sh
getLatestsStableVersions
```
to olm/prepare-community-operators-update.sh

File: olm/prepare-community-operators-update.sh [hunk 3]
```diff
@@ -114,56 +119,46 @@ do
...
       continue
     fi
-    lastPackagePreReleaseVersion=$(yq -r '.channels[] | select(.name == "'$channel'") | .currentCSV' "${sourcePackageFilePath}" | sed -e "s/${packageName}.v//")
-    lastPublishedPackageVersion=$(yq -r '.channels[] | select(.name == "'$channel'") | .currentCSV' "${destinationPackageFilePath}" | sed -e "s/eclipse-che.v//") 
-    if [[ $channel == "stable-all-namespaces" && -z $lastPublishedPackageVersion ]];then
-      lastPublishedPackageVersion=$lastPackagePreReleaseVersion
+
+    if [[ $channel == "stable-all-namespaces" ]];then
+      # Add suffix for stable-<all-namespaces> channel
+      LAST_PACKAGE_VERSION="$LAST_PACKAGE_VERSION-all-namespaces"
```
Comment (AndrienkoAleksandr) [inline olm/prepare-community-operators-update.sh:125]: ### 
```suggestion
      LAST_PACKAGE_VERSION="${LAST_PACKAGE_VERSION}-all-namespaces"
```

**Comment by AndrienkoAleksandr** [comment]
I think variable https://github.com/eclipse-che/che-operator/pull/1041/files#diff-87596dbd1244260e3d5bf59cc959c0aa3739edc345d88892fc7c6d8f3eaef2c8R115 is not used any more, so you can remove it.

**Review by AndrienkoAleksandr — APPROVED** [review]
Great job!

**Pull request merged**